File: 2022_03_cpp/220329.cpp
//
// Created by z on 3/29/2022.
// # 287
// Actually there are solutions with constant extra space, but I did not figure out.
// Sooner or later I will learn to use loop graph totally.

class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        unordered_map<int, int> m;
        for (auto &x: nums)
        {
            if (m.find(x) == m.end())
                m[x] = 1;
            else
                return x;
        }
        return -1;
    }
};

File: 2022/2022_python/220329.py
#
# created by z on 3/29/2022
# 287
# 220328.py doesn't exist

class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        m: dict = {}
        for x in nums:
            if m.get(x) == None:
                m[x] = 1
            else:
                return x
        return -1